packages/db/src/project/loadCompile/compilations.ts

```diff
@@ -168,19 +168,41 @@ function toSourceMapInputs(options: {
 function toImmutableReferencesInputs(options: {
   contracts: Contract[];
 }): DataModel.ImmutableReferenceInput[] {
-  const immutableReferences = options.contracts
-    .filter(({ immutableReferences }) => {
+  const immutableReferences: DataModel.ImmutableReferenceInput[] = options.contracts
+    .filter(contract => contract)
+    .filter(({ immutableReferences = {} }) => {
       return Object.keys(immutableReferences).length > 0;
     })
     .map(contract => {
-      return Object.entries(contract.immutableReferences).map(reference => {
-        return {
-          astNode: reference[0],
-          bytecode: contract.db.createBytecode,
-          length: reference[1][0].length,
-          offsets: reference[1].map(({ start }) => start)
-        };
-      });
+      return Object.entries(contract.immutableReferences)
+        .map(([astNode, slices]):
+          | DataModel.ImmutableReferenceInput
+          | undefined => {
+          // HACK start/length might actually be required, but contract-schema
+          // is wrong?
+          const definedSlices = slices.filter(
```

Comment:
In `contract-schema`, there's no `"required": ["start", "length"]`, so that slice definition gets converted to the TS type `{ start?: number; length?: number }`.

I'm assuming this is wrong, but I'm handling it here just to avoid the distraction (maybe this is worth opening an issue to track? let me know!)